src/main/java/gregtech/common/metatileentities/multi/electric/MetaTileEntityLargeMiner.java

```diff
@@ -0,0 +1,628 @@
+
+    private final Material material;
+    private final AtomicInteger x = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger y = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger z = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger startX = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger startZ = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger startY = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger tempY = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger mineX = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger mineZ = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger mineY = new AtomicInteger(Integer.MAX_VALUE);
+    private IEnergyContainer energyContainer;
+    protected IMultipleTankHandler inputFluidInventory;
+    protected IItemHandlerModifiable outputInventory;
+    private boolean isActive = true;
+    private boolean done = false;
+    private boolean silkTouch = false;
+    private boolean chunkMode = false;
+
+    private final LinkedList<BlockPos> blockPos = new LinkedList<>();
+    private int radius;
+    private int pipeLength = 0;
+    private boolean invFull = false;
+    private final int tier;
+    private int overclockAmount;
+    private final int chunkRadius;
+    private final int tick;
+    private final int drillingFluidConsumePerTick;
+    private final int fortune;
+    private final String romanNumeralString;
+
+    private static final Cuboid6 PIPE_CUBOID = new Cuboid6(4 / 16.0, 0.0, 4 / 16.0, 12 / 16.0, 1.0, 12 / 16.0);
+
+
+    public MetaTileEntityLargeMiner(ResourceLocation metaTileEntityId, int tier, Material material, int tick, int chunkRadius, int drillingFluidConsumePerTick, int fortune) {
+        super(metaTileEntityId);
+        this.tick = tick;
+        this.material = material;
+        this.tier = tier;
+        this.chunkRadius = chunkRadius;
+        this.radius = chunkRadius * 16;
+        this.drillingFluidConsumePerTick = drillingFluidConsumePerTick;
+        this.fortune = fortune;
+        this.romanNumeralString = GTUtility.romanNumeralString(fortune);
+        reinitializeStructurePattern();
+    }
+
+
+    @Override
+    public void invalidateStructure() {
+        super.invalidateStructure();
+        resetTileAbilities();
+        if (isActive)
+            setActive(false);
+    }
+
+    @Override
+    protected void formStructure(PatternMatchContext context) {
+        super.formStructure(context);
+        initializeAbilities();
+    }
+
+    private void initializeAbilities() {
+        this.inputFluidInventory = new FluidTankList(false, getAbilities(MultiblockAbility.IMPORT_FLUIDS));
+        this.outputInventory = new ItemHandlerList(getAbilities(MultiblockAbility.EXPORT_ITEMS));
```

Comment:
Are there notifiable versions of these that should be used? Maybe Trousers knows something

Reply:
I don't know of any notifiable ItemHandler or FluidTank List classes personally.